### PR TITLE
Implement Matrix2x2

### PR DESCRIPTION
**Pull request by hyp3rflow**
## Description
This PR includes below traits and its implementation for matrix2x2.
- fill
- set_diagonal
- row_length
- col_length
- get_row
- get_col
- transpose
- min
- abs_min
- max
- abs_max
- trace
- determinant
- inverse
- sum
- average
- make_zero
- make_identity
- make_rotation
- add
- sub
- mul
- div

### CODE REVIEW

File: oxide_math/matrix2x2.rs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    }
+    fn make_rotation(theta: f32) -> Matrix2x2 {
+        [[theta.cos(), -theta.sin()], [theta.sin(), theta.cos()]]
+    }
+    fn make_translation() -> Matrix2x2 {
+        panic!("TODO: implement")
+    }
+}
+
+impl Add for Matrix2x2 {
```
Comment (BeLeap) [inline oxide_math/matrix2x2.rs:131]: Why did you implemented it like this, not overloading operator?

**Review by LuneFleurs — APPROVED** [review]
Except Changseo's review, it looks good to me

File: oxide_math/matrix2x2.rs (hunk 1 above)
Comment (hyp3rflow) [reply oxide_math/matrix2x2.rs:131]: ops overload is not supported for type

**Review by utilForever — CHANGES_REQUESTED** [review]
Please write description and apply BeLeap's vector code.

**Review by BeLeap — COMMENTED** [review]
It would be nice to have a trait that binds the matrix together.

Can be used as below
```rust
fn rotate(&self, rotation_matrix: Box<Dyn matrix> {}
```

**Review by BeLeap — APPROVED** [review]

**Review by utilForever — APPROVED** [review]
I'll refactor this code after it is merged.

**Comment by hyp3rflow** [comment]
hooray!

**Pull request merged**